src/groupByKey.ts

```diff
@@ -2,6 +2,21 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
+export function groupByKey<
+  T, R extends keyof T
+>(
+  items: T[], key: R,
+): GroupsMap<T> {
   // write code here;
+  const result: GroupsMap<T> = {};
+
+  for (let i: number = 0; i < items.length; i += 1) {
```

Comment:
I don't think you need to type this. Some of the types are automatically infered
```suggestion
  for (let i = 0; i < items.length; i += 1) {
```